program-runtime/src/invoke_context.rs

```diff
@@ -796,75 +805,50 @@ impl<'a> InvokeContext<'a> {
         self.sysvar_cache
     }
 
+    // Should alignment be enforced during user pointer translation
+    pub fn get_check_aligned(&self) -> bool {
+        self.transaction_context
+            .get_current_instruction_context()
+            .and_then(|instruction_context| {
+                instruction_context.try_borrow_last_program_account(self.transaction_context)
+            })
+            .map(|program_account| *program_account.get_owner() != bpf_loader_deprecated::id())
+            .unwrap_or(true)
```

Comment:
This will default to true if try_borrow_last_program_account fails, which seems
like might or might not depending on the call site? I'd add at least a debug_assert